### PR TITLE
#686 Adds help information to gulp 

### PR DESCRIPTION
**Pull request by jaboc83**
All tasks are listed with help information using [gulp-help](https://www.npmjs.com/package/gulp-help).

I used the descriptions I found in the `DEVELOPING.md` file for most of it and made some guesses on the rest. 
2015-10-20 08_53_58-ubuntu_default_1441824855171_86120 running - oracle vm virtualbox

### CODE REVIEW

File: build-system/tasks/lint.js [hunk 1]
```diff
@@ -17,6 +17,7 @@
 
 var argv = require('minimist')(process.argv.slice(2));
 var config = require('../config');
+var gulp = require('gulp-help')(require('gulp'));
 var eslint = require('gulp-eslint');
 var gulp = require('gulp');
```
Comment (jridgewell) [inline build-system/tasks/lint.js]: `gulp` is redeclared here.

Comment (jaboc83) [inline build-system/tasks/lint.js]: Ahh, nice catch. Bad merge.

File: gulpfile.js [hunk 2]
```diff
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-var gulp = require('gulp');
+var gulp = require('gulp-help')(require('gulp'));
```
Comment (erwinmombay) [inline gulpfile.js]: sweet

**Comment by erwinmombay** [comment]
fixes https://github.com/ampproject/amphtml/issues/686

File: build-system/tasks/test.js [hunk 3]
```diff
@@ -72,4 +72,12 @@ gulp.task('test', ['build'], function(done) {
   }
 
   karma.start(config, done);
+}, {
+  options: {
+	'verbose': 'With logging enabled',
+	'watch': 'Watches for changes in files, runs corresponding test(s)',
+	'saucelabs': 'Runs test on saucelabs (requires setup)',
+	'safari': 'Runs tests in Safari',
+	'firefox': 'Runs tests in Firefox'
```
Comment (erwinmombay) [inline build-system/tasks/test.js]: de-indent by 2. will LGTM right after

**Comment by erwinmombay** [comment]
@jaboc83 , qq this still works with just `gulp.task('name', task)` right? just want to make sure people don't get tripped up if they're not aware of `gulp-help` getting installed. (i looked at the gulp-help code and that seems to be the case but just want to confirm with you.)

**Comment by jaboc83** [comment]
Yep that will still work. If you don't add a message it will just show up with a blank description like so: 
2015-10-20 08_53_58-ubuntu_default_1441824855171_86120 running - oracle vm virtualbox

**Comment by erwinmombay** [comment]
cool, thanks for confirming.

**Pull request merged**